services/self-service/src/main/java/com/epam/dlab/backendapi/service/impl/EndpointServiceImpl.java

```diff
@@ -68,13 +68,21 @@ public EndpointDTO get(String name) {
 				.orElseThrow(() -> new ResourceNotFoundException("Endpoint with name " + name + " not found"));
 	}
 
+	/**
+	 * Create new endpoint object in the System.
+	 * The EndPoint objects should contain Unique values of the 'url' and 'name' fields,
+	 * i.e two objects with same URLs should not be created in the system
+	 * @param userInfo user properties
+	 * @param endpointDTO object with endpoint fields
+	 */
 	@Override
 	public void create(UserInfo userInfo, EndpointDTO endpointDTO) {
+		if(endpointDAO.getEndpointWithUrl(endpointDTO.getUrl()).isPresent()) throw new ResourceConflictException("A Duplication of the Endpoint URL!");
```

Comment:
We do a call to mongo [here](https://github.com/apache/incubator-dlab/pull/542/files#diff-bd755c21a7b74a54525de3eeaea2efecR82) and we already have an endpoint entity. There is no need to call mongo again

Reply:
Also, please throw an exception inside `if` body

Reply:
The Exception has been thrown in the 'if' instruction.
The error messages has been updated according requirements